access/validator.go

```diff
@@ -259,6 +264,19 @@ func (v *TransactionValidator) checkAddresses(tx *flow.TransactionBody) error {
 	return nil
 }
 
+// every key (account, key index combination) can only be used once for signing
+func (v *TransactionValidator) checkSignatureDuplications(tx *flow.TransactionBody) error {
+	observedSigs := make(map[string]bool)
```

Comment:
yeah but that reduced the readability a lil bit - I like it when you can do
 `if observedSigs[keyStr] {}`

Reply:
yeah as far as I know they have minimal performance impact. and I like the readability aspect

Reply:
This is fine in this case. 
Using the `struct{}` is a [common](https://codetree.dev/golang-implementing-sets/) way to implement hash sets in Golang since there is no standard type `set`, but I agree it's not very readable.